File: src/main/java/tn/esprit/models/Commentaire.java
package tn.esprit.models;

import java.time.LocalDateTime;

public class Commentaire {
    private int idCommentaire;
    private int idPost;
    private String contenu;
    private LocalDateTime dateCreation;

    public Commentaire() {}

    public Commentaire(int idPost, String contenu) {
        this.idPost = idPost;
        this.contenu = contenu;
        this.dateCreation = LocalDateTime.now();
    }

    public int getIdCommentaire() { return idCommentaire; }
    public void setIdCommentaire(int idCommentaire) { this.idCommentaire = idCommentaire; }

    public int getIdPost() { return idPost; }
    public void setIdPost(int idPost) { this.idPost = idPost; }

    public String getContenu() { return contenu; }
    public void setContenu(String contenu) { this.contenu = contenu; }

    public LocalDateTime getDateCreation() { return dateCreation; }
    public void setDateCreation(LocalDateTime dateCreation) { this.dateCreation = dateCreation; }
}

File: src/Entity/Commentaire.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\CommentaireRepository;

#[ORM\Entity(repositoryClass: CommentaireRepository::class)]
#[ORM\Table(name: 'commentaire')]
class Commentaire
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    #[ORM\Column(type: 'integer')]
    private ?int $id_commentaire = null;

    public function getId_commentaire(): ?int
    {
        return $this->id_commentaire;
    }

    public function setId_commentaire(int $id_commentaire): self
    {
        $this->id_commentaire = $id_commentaire;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: false)]
    private ?string $contenu = null;

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): self
    {
        $this->contenu = $contenu;
        return $this;
    }

    #[ORM\Column(type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $date_commentaire = null;

    public function getDate_commentaire(): ?\DateTimeInterface
    {
        return $this->date_commentaire;
    }

    public function setDate_commentaire(?\DateTimeInterface $date_commentaire): self
    {
        $this->date_commentaire = $date_commentaire;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $id_post = null;

    public function getId_post(): ?int
    {
        return $this->id_post;
    }

    public function setId_post(int $id_post): self
    {
        $this->id_post = $id_post;
        return $this;
    }
}